components/batch-system/src/batch.rs

```diff
@@ -545,6 +545,7 @@ where
         let t = thread::Builder::new()
             .name(name)
             .spawn_wrapper(move || {
+                tikv_alloc::thread_allocate_exclusive_arena().unwrap();
```

Comment:
Does it mean memory arena can not be shared between threads?
Will it aggravate memory fragment or even OOM?
Can it tell how much memory is held (but not allocated) by a thread?

Reply:
- By default, the number of arena is four times of CPU cores. So currently, it's already nearly one arena per one thread. So I don't think it would aggravate memory fragment in TiKV case.

- What you mean by "held" is holding the memory allocated by other thread? Can't as allocator doesn't have the knowledge that the memory ownership is moved to another thread. At least, when it's freed in other threads, the memory usage accounted in the original allocated thread would be decreased.

Reply:
According to the [document](https://jemalloc.net/jemalloc.3.html), " There is a small fixed per-arena overhead, and additionally, arenas manage memory completely independently of each other, which means a small fixed increase in overall memory fragmentation. These overheads are not generally an issue, given the number of arenas normally used. Note that using substantially more arenas than the default is not likely to improve performance, mainly due to reduced cache performance. However, it may make sense to reduce the number of arenas if an application does not make much use of the allocation functions."

To make sure the overhead is small, can you run some read write mixed workload to compare the memory usage with and without this feature? @Connor1996

Reply:
There are four rounds of sysbench tests, read_only, insert, write_only, read_write case respectively. And there is no noticeable difference in memory usage
![07ce2f8c-43c6-4ca9-9d31-1e161785afbc](https://github.com/tikv/tikv/assets/13497871/d986ba13-a7ad-430a-ab1d-3dcd9ba92a8d)